### PR TITLE
update react-swipeable to v5

### LINKED ISSUE #321: Swiping through images in a gallery getting an error
Sorry - maybe all these errors are just me doing something wrong. But I set up a gallery and as I'm scrolling through the images, I get this error:

image

image

That's react-swipeable, not react-image-gallery, right? I just want to make sure I'm setting this up right before I ask a question elsewhere.

### PR DESCRIPTION
**Pull request by hartzis**
### update react-swipable to v5

- closes #321 
- closes https://github.com/dogfessional/react-swipeable/issues/128

##### TODOs:
- [x] match v4 functionality
- [x] use `dir` and try to clean up directional logic?

...
I found some time, 🎉 and I know we discussed possibly just removing `react-swipeable` in #321, but opened this to see what updating to v5 would look like.

### CODE REVIEW

**Comment by xiaolin** [comment]
Looks good so far!

**Comment by gcosgreave-us** [comment]
Is this going to be merged?

**Comment by hartzis** [comment]
@gcosgreave-us I got pulled away due to some bug fixes for [react-swipeable](https://github.com/dogfessional/react-swipeable/releases), but I do have a tentative plan to get back to this early next week. It is close.

**Comment by gcosgreave-us** [comment]
Awesome news @hartzis 👍

**Comment by hartzis** [comment]
@xiaolin I think this is ready for you to take a look.

- cleaned up a few directional checks with `react-swipeable`'s `dir` property.
- moved `event.preventDefault()` call into `react-image-gallery` via `react-swipeable` passing the event
  - allowed me to add check for `event.cancelable` to fix #321

**Comment by xiaolin** [comment]
@hartzis Sounds good, i'll take a look.

File: src/ImageGallery.jsx [hunk 1]
```diff
@@ -539,14 +541,20 @@ export default class ImageGallery extends React.Component {
...
+    if (
+      (this.props.preventDefaultTouchmoveEvent || scrollingLeftRight)
+      && event.cancelable
+    ) event.preventDefault();
     if (!isTransitioning && !scrollingUpDown) {
-      const side = deltaX < 0 ? 1 : -1;
+      const side = dir === RIGHT ? 1 : -1;
 
-      let offsetPercentage = (delta / galleryWidth * 100);
+      let offsetPercentage = (absX / galleryWidth * 100);
```
Comment (hartzis) [inline src/ImageGallery.jsx]: `delta` here in [react-swipeable v4](https://github.com/dogfessional/react-swipeable/tree/4.x#event-props) was actually `absX` so updated to use `absX` in v5.

**Comment by hartzis** [comment]
Just as a heads up, I was able to run the example and test everything I could and from my end. Let me know if you have questions/concerns/feedback.

**Comment by xiaolin** [comment]
LGTM, I'm going to do a round of testing on my end and merge.

**Comment by xiaolin** [comment]
@hartzis Looked through and everything seems to be working correctly. Thanks for the amazing PR!

**Pull request merged**